haskell/nettest/Permit.hs

```diff
@@ -129,32 +123,12 @@ permitScenario = uncapsNettest $ do
   callFrom (addressResolved user1) contract (Call @"Accept_ownership") ()
 
   where
-    createUser :: MonadNettest caps base m => Alias -> m (Alias, Address)
-    createUser alias = do
-      addr <- newAddress alias
+    createUser :: MonadNettest caps base m => AliasHint -> m (Alias, Address)
+    createUser aliasHint = do
+      addr <- newAddress aliasHint
       -- Note: `newAddress` prepends the alias with the prefix from the
       -- `MorleyClientConfig`.
       -- So we use `getAlias` to get the actual alias (prefix included)
       -- associated with this address.
-      actualAlias <- getAlias (AddressResolved addr)
-      pure (actualAlias, addr)
-
-    getChainId :: MonadNettest caps base m => m ChainId
-    getChainId = do
-      addr <- originateSimple "chainIdContract" "" chainIdContract
-      call addr CallDefault ()
-
-      storage <- fromVal @ByteString <$> getStorage @(ToT ByteString) (addressResolved addr)
-      case lUnpackValue @ChainId storage of
-        Left err -> error $ "Invalid ChainId retrieved from 'chainIdContract':\n" <> pretty err
-        Right chainId -> pure chainId
-
-
--- | A contract that stores the current ChainId in its storage.
---
--- NOTE: `compileLorentzContract` requires the contract's parameter to have an instance for
--- `HasAnnotation`, and ChainId doesn't.
--- So, instead, we store it as a ByteString and pack/unpack as needed.
-chainIdContract :: Contract () ByteString
-chainIdContract = defaultContract $
-  L.drop # L.chainId # L.pack # L.nil # L.pair
+      alias <- getAlias (AddressResolved addr)
+      pure (alias, addr)
```

Comment:
Oh, apparently it resolves https://github.com/tqtezos/stablecoin/issues/109

Reply:
Yup, I added #109 to this PR's description now :)